Add 0.15 release blog (#483)
kserve/website: docs/blog/articles/2025-05-27-KServe-0.15-release.md

[Envoy Gateway](https://gateway.envoyproxy.io) is also now supported in KServe along with [Kubernetes Gateway API](https://gateway-api.sigs.k8s.io). Unlike traditional gateway solutions, Envoy AI Gateway provides advanced capabilities tailored to AI serving, including:

- Dynamic model routing based on request content, model metadata, or user context.
- Built-in support for multi-tenant inference, with fine-grained access controls and authentication.
- Unified API for routing and managing LLM/AI traffic easily.
- Integrated observability for model-level performance insights.
- Extensibility for inference-specific policies like rate-limiting by token, and model lifecycle management.
- Automatic failover mechanisms to ensure service reliability.

This integration enables a unified, intelligent entrypoint for both predictive and generative workloads—scaling from traditional models to complex LLMs—all while abstracting infrastructure complexity from the user. Please refer to [Envoy AI Gateway integration doc](https://kserve.github.io/website/0.15/admin/ai-gateway_integration) for more details.

---

### Multi-Node Inference

> KServe has implemented an initial version of multi-node, multi-GPU inference when your model is too large to fit in a single node such as Llama 3.1 405B. As models continue to grow, multi-node inference capabilities are increasingly important for production deployments that require real-time user experience. Please refer to the [Multi Node inference doc](https://kserve.github.io/website/0.15/modelserving/v1beta1/llm/huggingface/multi-node) for more details.

Reviewer: ```suggestion
To support LLMs too large for a single node (e.g., Llama 3.1 405B), KServe v0.15 introduces multi-node inference across distributed GPUs, unlocking large model serving at scale. As models continue increase in size, multi-node inference capabilities are increasingly important for production deployments that require real-time user experience. Please refer to the [Multi Node inference doc](https://kserve.github.io/website/0.15/modelserving/v1beta1/llm/huggingface/multi-node) for more details.
```

The community is also working on a [new distributed inference API](https://github.com/kserve/kserve/issues/4433) to allow scaling Multi Node Inference and support Disaggregatd Prefilling which is targeted for large LLM deployments. 

---

> ### LLM Autoscaler with KEDA

Reviewer: ```suggestion
### LLM Autoscaler with KEDA [(Kubernetes Event-driven Autoscaling)](https://keda.sh/)
```

Autoscaling Large Language Models (LLMs) presents a complex set of challenges due to their immense computational demands and unique architectural characteristics. The dynamic nature of LLM inference, with varying input lengths and token generation speeds, further complicates the prediction of resource needs, demanding sophisticated and adaptive autoscaling solutions. KServe now integrates with [**KEDA**](https://keda.sh) (Kubernetes Event-Driven Autoscaling) offers a powerful solution to many of the challenges associated with LLM autoscaling by extending Kubernetes' native Horizontal Pod Autoscaler (HPA) capabilities. KEDA can monitor custom metrics which means you can expose LLM metrics from your LLM inference servers and use KEDA to scale based on these precise indicators.

> 

Reviewer: ```suggestion
This allows you to:
* Scale based on token usage, input size, or generation rate.
* Fine-tune compute allocation for cost-efficiency and performance.
* Remove dependency on Knative for autoscaling.
```

---

### Distributed KV Cache with LMCache

Key-Value (KV) cache offloading is a technique used in large language model (LLM) serving to store and reuse the intermediate key and value tensors generated during model inference. In transformer-based models, these KV caches represent the context for each token processed, and reusing them allows the model to avoid redundant computations for repeated or similar prompts.

By enabling KV cache offloading across multiple requests and serving instances, you can achieve reduced TTFT, improved scalablity for shared cache across replicas and improved user experience for multi-turn QA or RAG.

KServe integrates [LMCache](https://lmcache.ai), the-state-of-art KV cache layer library developped by LMCache Lab to reduce inference costs and ensure SLOs for both latency and throughput at scale. Please follow the [LMCache integration doc](https://kserve.github.io/website/0.15/modelserving/v1beta1/llm/huggingface/kv_cache_offloading/#overview) to optimize your GenAI inference workload.

---

### Advanced Model Caching Mechanisms

> To address the challenges of serving large models, KServe v0.15 introduces advanced model caching features:

Reviewer: ```suggestion
To reduce model loading times and improve overall efficiency of serving large models, KServe v0.15 introduces advanced model caching features:
```

- **LocalModelCache Enhancements:** Improved the LocalModelCache custom resource to support multiple node groups, providing greater flexibility in model placement and caching strategies.
- **Node Agent Improvements:** Enhanced the local model node agent for better performance and reliability.

> These enhancements reduce model loading times and improve the overall efficiency of serving large models.

Reviewer: ```suggestion
```
